### DISCUSSION TITLE
MPI error loading mesh

### DISCUSSION THREAD

**Discussion by rebpoli** [libMesh/libmesh#3187] [Q&A]
Hi

I get the error below while loading a gmsh file.
I have no clue what is going on, I am wondering if anyone have seen this before.

Thanks,
Renato

PETSC: Attaching gdb to ./sim of pid 202202 on display localhost:11.0 on machine sdumont15
[sdumont15:202189] 11 more processes have sent help message help-mpi-btl-openib.txt / ib port not selected
[sdumont15:202189] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
[sdumont15:202189] 9 more processes have sent help message help-mpi-btl-openib.txt / error in device init
[sdumont15:202189] 1 more process has sent help message help-mpi-btl-openib.txt / no active ports found

(gdb) 
#0  0x00007fd337fa83bf in clock_nanosleep () from /usr/lib/x86_64-linux-gnu/libc.so.6
#1  0x00007fd337fae047 in nanosleep () from /usr/lib/x86_64-linux-gnu/libc.so.6
#2  0x00007fd337fadf7e in sleep () from /usr/lib/x86_64-linux-gnu/libc.so.6
#3  0x00007fd3386952c0 in PetscSleep () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#4  0x00007fd3386bff55 in PetscAttachDebugger () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#5  0x00007fd3386bffe7 in PetscAttachDebuggerErrorHandler () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#6  0x00007fd3386c11c5 in PetscError () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#7  0x00007fd3386c7730 in PetscSignalHandlerDefault () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#8  0x00007fd3386c741d in PetscSignalHandler_Private () from /opt/petsc/arch-linux-c-opt/lib/libpetsc.so.3.16
#9  <signal handler called>
#10 0x00007fd2da59f1c2 in ompi_mtl_ofi_progress_no_inline () from /usr/lib/x86_64-linux-gnu/openmpi/lib/openmpi3/mca_mtl_ofi.so
#11 0x00007fd2ec85e854 in opal_progress () from /lib/x86_64-linux-gnu/libopen-pal.so.40
#12 0x00007fd338496e25 in ompi_request_default_wait_all () from /usr/lib/x86_64-linux-gnu/libmpi.so.40
#13 0x00007fd3384ebd4e in ompi_coll_base_bcast_intra_generic () from /usr/lib/x86_64-linux-gnu/libmpi.so.40
#14 0x00007fd3384ec061 in ompi_coll_base_bcast_intra_pipeline () from /usr/lib/x86_64-linux-gnu/libmpi.so.40
#15 0x00007fd2d9bbbdae in ompi_coll_tuned_bcast_intra_dec_fixed () from /usr/lib/x86_64-linux-gnu/openmpi/lib/openmpi3/mca_coll_tuned.so
#16 0x00007fd3384aeb10 in PMPI_Bcast () from /usr/lib/x86_64-linux-gnu/libmpi.so.40
#17 0x00007fd33bb8a9b3 in libMesh::MeshCommunication::broadcast(libMesh::MeshBase&) const () from /usr/local/lib/libmesh_opt.so.0
#18 0x00007fd33bcea168 in libMesh::NameBasedIO::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /usr/local/lib/libmesh_opt.so.0
#19 0x00007fd33bd39d8d in libMesh::UnstructuredMesh::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void*, bool, bool) () from /usr/local/lib/libmesh_opt.so.0
#20 0x0000559b742f30f5 in MeshConfig::load_mesh (this=this@entry=0x7ffc244a32e0, mesh=...) at /petrobr/parceirosbr/chimasict/_chimas_mirror/_renato-linux_renato/chimas3d/src/MeshConfig.cpp:31
#21 0x0000559b7429a63d in TimeLoop::main_loop (this=0x7ffc244a37e0) at /petrobr/parceirosbr/chimasict/_chimas_mirror/_renato-linux_renato/chimas3d/src/TimeLoop.cpp:50
#22 0x0000559b7429bf79 in TimeLoop::TimeLoop (this=0x7ffc244a37e0, ci=..., m=...) at /petrobr/parceirosbr/chimasict/_chimas_mirror/_renato-linux_renato/chimas3d/src/TimeLoop.cpp:25
#23 0x0000559b74288a77 in main (argc=<optimized out>, argv=<optimized out>) at /petrobr/parceirosbr/chimasict/_chimas_mirror/_renato-linux_renato/chimas3d/src/main.cpp:20

**Comment by jwpeterson** [comment]
Assuming you know that MPI is working on your system (you have run other sims in parallel successfully) I'd recompile/run in debug mode as you may get an assert before you hit this error down in MPI.
